### PR TITLE
Fix linux compilation

### PR DESCRIPTION
**Pull request by rjanvier**
Hi,
cc @dgirardeau 
Just a PR to fix Linux compilation. I did not made a full review of the code but just tried to make the code compile on Linux.
I think it makes code more corrects on some aspects. 

For the details, `break` inside an openMP loop is not correct (even if allowed by MSVC but the [doc itself says it's not correct)](https://learn.microsoft.com/en-us/cpp/parallel/concrt/convert-an-openmp-loop-that-uses-cancellation?view=msvc-170). Also assigning your `error` bool inside a parallel loop could leads to a race condition. I use an atomic but it could be an openMP `critical section`.

### CODE REVIEW

File: include/qVoxFallTools.h [hunk 1]
```diff
@@ -50,13 +52,13 @@ class qVoxFallTransform
 		const Vector3Tpl<float> Y(-std::sin(zRot), std::cos(zRot), 0);
 		const Vector3Tpl<float> Z(0, 0, 1);
 		const Vector3Tpl<float> Tr(0, 0, 0);
-		matrix = &ccGLMatrix::ccGLMatrix(X, Y, Z, Tr);
+		matrix = new ccGLMatrix(X, Y, Z, Tr);
```
Comment (dgirardeau) [inline include/qVoxFallTools.h]: don't we need a destructor to free these matrices now?

Comment (dgirardeau) [reply include/qVoxFallTools.h]: Or maybe convert these ccGLMatrix* pointers to plain instances (the impact on the code might be bigger though).

File: src/qVoxFallTools.cpp [hunk 2]
```diff
@@ -67,7 +67,7 @@ ccBox* qVoxFallTransform::CreateVoxelMesh(CCVector3 V, float voxelSize, int voxe
 	const Vector3Tpl<float> X(1, 0, 0);
 	const Vector3Tpl<float> Y(0, 1, 0);
 	const Vector3Tpl<float> Z(0, 0, 1);
-	ccGLMatrix* matrix = &ccGLMatrix::ccGLMatrix(X, Y, Z, V);
+	ccGLMatrix* matrix = new ccGLMatrix(X, Y, Z, V);
```
Comment (dgirardeau) [inline src/qVoxFallTools.cpp]:
```suggestion
	ccGLMatrix matrix(X, Y, Z, V);

```

File: src/qVoxFallProcess.cpp [hunk 3]
```diff
@@ -439,7 +442,7 @@ bool qVoxFallProcess::Compute(const qVoxFallDialog& dlg, QString& errorMessage,
 	s_VoxFallParams.isEmptyBefore.resize(voxelGrid.innerCellCount(), true);
 	if (s_VoxFallParams.exportBlocksAsMeshes)
 	{
-		s_VoxFallParams.clusters.resize(voxelGrid.innerCellCount(), NULL);
+		s_VoxFallParams.clusters.resize(voxelGrid.innerCellCount(), 0);
```
Comment (dgirardeau) [inline src/qVoxFallProcess.cpp:445]: or nullptr? (haven't checked what the type is, sorry)

**Review by dgirardeau — CHANGES_REQUESTED** [review]

File: src/qVoxFallProcess.cpp (hunk 3 above)
Comment (rjanvier) [reply src/qVoxFallProcess.cpp:445]: it's a `std::vector<int> `so 0 should be fine.

File: include/qVoxFallTools.h (hunk 1 above)
Comment (rjanvier) [reply include/qVoxFallTools.h]: Thanks. I Thought it was already done into the object destructor.
Yes, plain instance should make more sense, I have to track usage of these matrices in the other parts of the code.

**Review by dgirardeau — APPROVED** [review]

**Pull request merged**